### PR TITLE
user authentication

### PR DESCRIPTION
**Pull request by HarshnaNHaswani**
please review

### CODE REVIEW

**Comment by ashifkhn** [comment]
Good to go.

File: src/components/RequiresAuth/index.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  const location = useLocation();
+  const {
+    authData: { isAuth },
+    error,
+    loadState
+  } = useAuth();
+  return isAuth ? (
+    <Outlet />
+  ): error?<h4>{error}</h4>: loadState ? <p>Loading...</p> : (
+    <Navigate to="/auth/login" state={{ from: location }} replace />
```
Comment (ivinayakg) [inline src/components/RequiresAuth/index.tsx:13]: great for ading the location

File: src/pages/Auth/Login.tsx [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    password: TextType;
+  };
+  const location = useLocation();
+  const locationState = location.state as { from: Location };
+
+  const testCreds = {
+    email: "guest@xyz.com",
+    password: "abc123hh",
+  };
+  const [status, setStatus] = useState({ isLoading: false, error: "" });
```
Comment (ivinayakg) [inline src/pages/Auth/Login.tsx:25]: instead of an object create a string based status

**Review by ivinayakg — APPROVED** [review]

File: src/pages/Auth/Login.tsx (hunk 2 above)
Comment (HarshnaNHaswani) [reply src/pages/Auth/Login.tsx:25]: Do you mean like
```js
status='loading'
status='error'
```
?

**Comment by HarshnaNHaswani** [comment]
thank you @ashifkhn @ivinayakg !

**Pull request merged**